src/python/pants/backend/jvm/tasks/scaladoc_gen.py

```diff
@@ -59,9 +64,46 @@ def create_scaladoc_command(self, classpath, gendir, *targets):
 
     args.extend(sources)
 
+    # Check the size of the ScalaDoc command arguments, and use a options file if too long.
+    #
+    # There are a couple practical limits to consider.
+    #
+    # - The length of any individual value must be less than MAX_ARG_STRLEN
+    #   which is hardcoded to 131072 bytes.
+    #
+    # - The total length of the command must be less than the OS's `getconf ARG_MAX`
+    #   value.
+    #
+    #   This value will vary by system. For exmaple, on my MacOS 10.13.6 Twitter-issued
+    #   developer laptop this value is currently 262144. Doing some research on the
+    #   internet shows a wide range of default values for this ranging from 131072 to
+    #   2097152 (2MB). Unfortunately, there doesn't seem to be an easy way to query the
+    #   system for this value. It's not in sysconfig.get_config_vars, or posix.environ,
+    #   or os.environ. We could shell out to `getconf` to get the actual value, but
+    #   anecdotal evidence suggests that there are other layers of limits beyond just
+    #   this POSIX value. POSIX requires ARG_MAX to be at least 4096 bytes, but it's
+    #   extremely unlikely to find a system with that low setting. Rather, I think a
+    #   practical middle ground would be the MAX_ARG_STRLEN fixed value.
+    #
+    #   So, given that rationale, if the arg list is longer than MAX_ARG_STRLEN
+    #   we should put it in an external file, just in case.
```

Comment:
Is there a reason we wouldn't want to dump everything into an argfile on every invocation? It would seem to remove the guesswork here, unless using an argfile removes some other functionality.

Reply:
Could use an arg file for every invocation, but it would be unnecessary for most invocations. Running this with `--combined` across large dependency graphs is kind of a "large Scala monorepo"-specific problem. I'm not sure how Pant's user base looks. Are most users in large monorepos with lots of Scala? How commonly is this task used in this manner?

Not that there's much harm in using an arg file when it's not needed. I was just following the principle of least change (e.g. change the behaviour as little as possible to get the desired effect).

Reply:
I agree with @cosmicexplorer : it's nice to try and avoid this, but I don't think it's worth the extra codepath, so would just go with the argfile approach by default.

Reply:
The way I usually think about these situations is that command-line tools offer interfaces for a user as well as interfaces for other programs -- while using arguments on the actual command line is likely the more common and preferred way to invoke this tool by hand, in Pants, we can feel free to use approaches such as arg files, or maybe config files generated on the fly, etc.